### PR TITLE
feature: move wpf out of main package to fix unit testing story on mono and unblock other ui frameworks

### PR DESCRIPTION
**Pull request by jkoritzinsky**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
This change moves the WPF support out of the main package and into a new `reactiveui-wpf` package. This enables multiple avenues, including:
 * No need for a separate Mono build to support Mono
 * `reactiveui-winforms` has no dependency on WPF assemblies
 * UI Frameworks on Mono or .NET Framework can use ReactiveUI without having to maintain their own forks or custom builds that patch out WPF service hook-ins. (i.e. Avalonia).

**What is the current behavior? (You can also link to an open issue here)**
#925. Currently, ReactiveUI assumes by default that an application referencing it on .net 4.5 wants to use WPF. Since WPF is not available on Mono, current ReactiveUI packages cannot be used on Mono platforms. Also, UI frameworks like Avalonia have to maintain their own forks of ReactiveUI to patch out code that might cause the runtime to load any WPF types.

**What is the new behavior (if this is a feature change)?**
The core ReactiveUI package can be used on the .NET Framework or Mono without requiring referencing WPF at all.

**Does this PR introduce a breaking change?**

💥💥💥 

This PR is a breaking change and will require documentation in the RELEASENOTES to instruct users of WPF to install the `reactiveui-wpf` and `reactiveui-events-wpf` packages.  `reactiveui` package no longer contains any WPF references.

💥💥💥

**Please check if the PR fulfills these requirements**
- [ x ] The commit follows our guidelines: https://github.com/reactiveui/reactiveui#contribute
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:
I don't know where to update the docs. Please tell me where to look and I will update them as needed!

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
@ghuntley Can you review this?

File: src/ReactiveUI.Wpf/ReactiveUI.Wpf.csproj [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  <PropertyGroup>
+    <TargetFramework>net45</TargetFramework>
+    <DefineConstants>$(DefineConstants);WPF</DefineConstants>
+    <Description>WPF specific extensions to ReactiveUI</Description>
+    <PackageId>reactiveui-wpf</PackageId>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="..\ReactiveUI\CreatesCommandBinding.cs" Link="CreatesCommandBinding.cs" />
+    <Compile Include="..\ReactiveUI\Platforms\windows-common\ActivationForViewFetcher.cs" Link="windows-common/ActivationForViewFetcher.cs" />
```
Comment (olevett) [inline src/ReactiveUI.Wpf/ReactiveUI.Wpf.csproj]: Sad that this needs individual file references :(

Comment (olevett) [inline src/ReactiveUI.Wpf/ReactiveUI.Wpf.csproj]: This shows up with a little cross in VS for me (I don't think VS really likes it!) - long term, I'd be pro splitting the implementations (maybe sharing a base class for common functionality so we don't have the mess of #ifdefs) but for now, I think these might be neater just being included in the root of the project (the link should be fairly easy to spot, and the windows-common doesn't make a huge amount of difference within the context of the WPF project)

File: src/ReactiveUI/CreatesCommandBinding.cs [hunk 2]
```diff
@@ -4,15 +4,20 @@
...
 
 #if NETFX_CORE
 using Windows.UI.Xaml;
 using Windows.UI.Xaml.Input;
+#elif WPF
+using System.Windows;
 #endif
 
+#if WPF
+namespace ReactiveUI.Wpf
```
Comment (olevett) [inline src/ReactiveUI/CreatesCommandBinding.cs]: I'd be tempted at this point to just pull the WPF copy into this project, and then remove the various `if`s from the shared copy. I think it's a bit neater than redefining the namespace

Comment (jkoritzinsky) [inline src/ReactiveUI/CreatesCommandBinding.cs]: The `RoutedEventArgs` type is in the WPF assemblies, so including the `Click` event as a default event won't work unless we use the `EventArgs` type.

**Comment by ghuntley** [comment]
Couple changes were required:
* Each package needs to be white-listed in `build.cake` otherwise the version won't be pinned or published to MyGet (or NuGet)
* Also needed to split out of `reactiveui-events`

This PR is a breaking change and will require documentation in the RELEASENOTES to instruct users of WPF to install the `reactiveui-wpf` and `reactiveui-events-wpf` packages. The `reactiveui` package no longer contains any net45 assemblies.

**Comment by ghuntley** [comment]
@jkoritzinsky thank-you very much for doing this. Fixes #925

**Pull request merged**